### PR TITLE
[UED-1160] [Add cookie consent dialog for milvus.io]

### PR DESCRIPTION
**Pull request by iynewz**

### CODE REVIEW

**Comment by shanghaikid** [comment]
Can you provide some screenshot?

File: src/components/milvusCookieConsent/index.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          buttonClasses={styles.button}
+          buttonWrapperClasses={styles.buttonWrapper}
+          expires={150}
+          onAccept={() => {
+            // alert("consent accepted");
+            EnableAnalytics();
+          }}
+        >
+          <div className={styles.textContainer}>
+            <p className={styles.headerContent}>How we use cookies</p>
```
Comment (nameczz) [inline src/components/milvusCookieConsent/index.jsx:65]: need i18n for the hard code text. Support cn and en.

File: src/components/milvusCookieConsent/index.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            <p className={styles.textContent}>
+              By continuing to browse or by clicking ‘Accept’, you agree to the
```
Comment (nameczz) [inline src/components/milvusCookieConsent/index.jsx:67]: same here

**Comment by iynewz** [comment]
> Can you provide some screenshot?

Sure, here are how the consent bar looks like on different screen sizes:
image
image
image
image
image

File: src/components/milvusCookieConsent/index.jsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  return typeof window !== "undefined";
+}
+
+function getValue(key, defaultValue) {
+  return isBrowser() && window.localStorage.getItem(key)
+    ? JSON.parse(window.localStorage.getItem(key))
+    : defaultValue;
+}
+
+function setValue(key, value) {
```
Comment (nameczz) [inline src/components/milvusCookieConsent/index.jsx:17]: rename to setLocalStorage is more clear to me

File: src/components/milvusCookieConsent/index.jsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import CookieConsent from "react-cookie-consent";
+import * as styles from "./milvusCookieConsent.module.less";
+import { useLocation } from "@reach/router";
+import { initializeAndTrack } from "gatsby-plugin-gdpr-cookies";
+
+function isBrowser() {
+  return typeof window !== "undefined";
+}
+
+function getValue(key, defaultValue) {
```
Comment (nameczz) [inline src/components/milvusCookieConsent/index.jsx:11]: rename to getLocalStorage is more clear to me

File: src/components/milvusCookieConsent/index.jsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  return isBrowser() && window.localStorage.getItem(key)
+    ? JSON.parse(window.localStorage.getItem(key))
```
Comment (nameczz) [inline src/components/milvusCookieConsent/index.jsx:13]: JSON.parse will cause some error when param is not json, we may need to use try catch for JSON.parse

File: src/components/milvusCookieConsent/index.jsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  }, [key, value]);
+
+  return [value, setter];
+}
+
+const MilvusCookieConsent = () => {
+  const location = useLocation();
+  if (isBrowser()) {
+    initializeAndTrack(location);
+  }
```
Comment (nameczz) [inline src/components/milvusCookieConsent/index.jsx:37]: should we need useEffect here

File: src/components/milvusCookieConsent/index.jsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+  const [bannerHidden, setBannerHidden] = useStickyState(
+    false,
+    "consentCookieHidden"
+  );
+
+  const EnableAnalytics = () => {
```
Comment (nameczz) [inline src/components/milvusCookieConsent/index.jsx:44]: The first letter of the normal function name should be lowercase. Only react function component or class is capital.

File: src/components/milvusCookieConsent/index.jsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    ? JSON.parse(window.localStorage.getItem(key))
+    : defaultValue;
+}
+
+function setValue(key, value) {
+  window.localStorage.setItem(key, JSON.stringify(value));
+}
+
+function useStickyState(defaultValue, key) {
+  const [value, setter] = React.useState(() => {
```
Comment (nameczz) [inline src/components/milvusCookieConsent/index.jsx:22]: For useState, normally we use like [value,setValue] is more clear for us to know what value we set.

**Review by nameczz — CHANGES_REQUESTED** [review]

**Review by nameczz — APPROVED** [review]

**Pull request merged**